### PR TITLE
UIDATIMP-746: Validation for default data typed into a field mapping profile is not working

### PR DESCRIPTION
**Pull request by IhorKoba**
## Overview

When default data (non-MARC field) is added to a field mapping profile, it must be enclosed in quotation marks. If the value is selected from a dropdown, the quotation marks are added automatically. If the value is typed by the user, no quotation marks are added, and no validation error appears.

## Approach

- Add `validateDefaultDataTypedInField` function
- Update `validateForQuotedStringOrMarcPathPattern` function
- Create `WithValidation` wrapper component
- Wrap all needed field with current component

## Refs

https://issues.folio.org/browse/UIDATIMP-746

### CODE REVIEW

File: src/utils/formValidators.js [hunk 1]
```diff
@@ -110,44 +110,42 @@ export const validateValueLength1 = value => validateValueLength(value, 1);
...
+ * Validate MARC path, quoted string and else condition. Match `910`, `910$a`, `"text"`, `LDR`, `LDR/7`, `005/7-10`,
  * `910$a "text"`, `910$a; else "text"; else 910`,
  * validateMARCWithElse('###REMOVE###', true)
  * // => null
  * @param value
  * @param {boolean} isRemoveValueAllowed
  * @returns {null|*}
  */
-export const validateMARCWithElse = (value, isRemoveValueAllowed) => {
+const validateForQuotedStringOrMarcPathPattern = (value, isRemoveValueAllowed) => {
```
Comment (mariia-aloshyna) [inline src/utils/formValidators.js:121]:
```suggestion
const validateQuotedStringOrMarcPathPattern = (value, isRemoveValueAllowed) => {
```

File: src/utils/formValidators.js [hunk 2]
```diff
@@ -110,44 +110,42 @@ export const validateValueLength1 = value => validateValueLength(value, 1);
...
 /**
- * Validate MARC path, quoted string and else condition. Match `910`, `910$a`, `"text"`,
+ * Validate MARC path, quoted string and else condition. Match `910`, `910$a`, `"text"`, `LDR`, `LDR/7`, `005/7-10`,
  * `910$a "text"`, `910$a; else "text"; else 910`,
  * validateMARCWithElse('###REMOVE###', true)
  * // => null
  * @param value
  * @param {boolean} isRemoveValueAllowed
  * @returns {null|*}
  */
```
Comment (mariia-aloshyna) [inline src/utils/formValidators.js:120]: Fix doc, please.

```suggestion
 * Validate MARC path, quoted string and else condition. Match `910`, `910$a`, `"text"`, `LDR`, `LDR/7`, `005/7-10`,
 * `910$a "text"`, `910$a; else "text"; else 910`,
 *
 * @param value
 * @param {boolean} isRemoveValueAllowed
 * @returns {boolean}
 *
 * @example
 *
 * validateForQuotedStringOrMarcPathPattern('###REMOVE###', true)
 * // => true
 *
 * validateForQuotedStringOrMarcPathPattern('TEST', false)
 * // => false
 */
```

File: src/utils/formValidators.js [hunk 3]
```diff
@@ -110,44 +110,42 @@ export const validateValueLength1 = value => validateValueLength(value, 1);
...
     `(((; else )(?=${quotedStringOrMarcPathPattern}))`,
     `((?<=(; else ))${quotedStringOrMarcPathPattern})))*$`].join(''));
 
-  if (isEmpty(value) || value === allowedValue || value.match(pattern)) {
-    return null;
-  }
+  return isEmpty(value) || value === allowedValue || value.match(pattern);
+};
+
+export const validateMARCWithElse = (value, isRemoveValueAllowed) => {
```
Comment (mariia-aloshyna) [inline src/utils/formValidators.js:135]: Add doc for this function, please.

File: src/utils/formValidators.js [hunk 4]
```diff
@@ -110,44 +110,42 @@ export const validateValueLength1 = value => validateValueLength(value, 1);
...
- * validateTextFieldRemoveValue('###REMOVE###')
- * // => null
- * @param value
- * @returns {null|*}
- */
-export const validateTextFieldRemoveValue = value => {
-  if (!isEmpty(value) && value.includes(REMOVE_OPTION_VALUE) && value !== REMOVE_OPTION_VALUE) {
-    return <FormattedMessage id="ui-data-import.validation.syntaxError" />;
-  }
+export const validateDefaultDataTypedInField = (value, isRemoveValueAllowed) => {
```
Comment (mariia-aloshyna) [inline src/utils/formValidators.js:143]:
```suggestion
export const validateQuotedStringOrMarcPath = (value, isRemoveValueAllowed) => {
```

Comment (mariia-aloshyna) [inline src/utils/formValidators.js:143]: Add doc for this function, please.

File: src/utils/formValidators.js (hunk 1 above)
Comment (IhorKoba) [reply src/utils/formValidators.js:121]: done

File: src/utils/formValidators.js (hunk 4 above)
Comment (IhorKoba) [reply src/utils/formValidators.js:143]: done

File: src/utils/formValidators.js (hunk 2 above)
Comment (IhorKoba) [reply src/utils/formValidators.js:120]: done

File: src/utils/formValidators.js (hunk 3 above)
Comment (IhorKoba) [reply src/utils/formValidators.js:135]: done

File: src/utils/formValidators.js (hunk 4 above)
Comment (IhorKoba) [reply src/utils/formValidators.js:143]: done

**Review by mariia-aloshyna — APPROVED** [review]

**Review by IvanKryzhanovskyi — APPROVED** [review]

**Pull request merged**